src/libYARP_dev/src/devices/ControlBoardWrapper/RPCMessagesParser.cpp

```diff
@@ -156,7 +156,7 @@ void RPCMessagesParser::handleControlModeMsg(const yarp::os::Bottle& cmd,
 {
     if (ControlBoardWrapper_p->verbose())
         yDebug("Handling IControlMode message\n");
-    if (! (rpc_iCtrlMode || rpc_iCtrlMode2) )
+    if ( !(rpc_iCtrlMode) || !(rpc_iCtrlMode2) )
```

Comment:
I think this is not the problem here, but the `*ok=false` a couple of lines later
Also this is changing the logic here (if I'm not wrong)

| A | B | !(A or B) | !A or !B |
|---|---|---------|----------|
| 1 | 1 | 0       | 0        |
| 1 | 0 | **0** | **1**   |
| 0 | 1 | **0** | **1**   |
| 0 | 0 | 1       | 1        |

Reply:
The problem is that in the code following this `if`, `rpc_iCtrlMode` and `rpc_iCtrlMode2` were dereferenced many times without a check, so I thought both `rpc_iCtrlMode` and `rpc_iCtrlMode2` was necessary to this piece of code. But now I see this is not true and the file analyzed by coverity does not match the one on the master were every missing check has already been done by the clang warning fixings.